src/util/schnorr.rs

```diff
@@ -108,6 +171,24 @@ impl TweakedPublicKey {
     }
 }
 
+impl TweakedKeyPair {
+    /// Creates a new [`TweakedKeyPair`] from a [`KeyPair`]. No tweak is applied, consider
+    /// calling `tap_tweak` on an [`UntweakedKeyPair`] instead of using this constructor.
+    ///
+    /// This method is dangerous and can lead to loss of funds if used incorrectly.
+    /// Specifically, in multi-party protocols a peer can provide a value that allows them to steal.
+    #[inline]
+    pub fn dangerous_assume_tweaked(pair: KeyPair) -> TweakedKeyPair {
+        TweakedKeyPair(pair)
+    }
+
+    /// Returns the underlying key pair
+    #[inline]
+    pub fn into_inner(self) -> KeyPair {
+        self.0
+    }
```

Comment:
Should we also `impl From<TweakedKeyPair> for KeyPair`?
Also maybe `into_raw` is a clearer name?

Reply:
Will do a From impl in the follow-up PR